packages/react-native/ReactCommon/react/renderer/mounting/ShadowTree.cpp

```diff
@@ -365,6 +365,32 @@ CommitStatus ShadowTree::commit(
   }
 }
 
+ShadowNode::Unshared dirtyTreeRecursive(const std::shared_ptr<const ShadowNode> node) {
```

Comment:
A recursion on the tree could be expensive. I'm a bit afraid of the perf implications of this. 
Can we look if there are other ways to invalidate the layout node or if we have to do something similar somewhere else that is perhaps a bit more efficient?

Also, this is a part of the codebase I don't know well yet, so we might want to wait for some other people to chime in here.

Reply:
I believe one thing that could be done to optimize this is to only clone paths where either `TextInput` or `Text` is the leaf node, though it would still clone a large part of the tree (and I don't think logic like that exists yet).

The problem with that approach would be that any custom native component with text that would be affected by the change in font scale would be skipped.